rosbag2_compression/src/rosbag2_compression/sequential_compression_writer.cpp

```diff
@@ -62,6 +69,42 @@ SequentialCompressionWriter::~SequentialCompressionWriter()
 
 void SequentialCompressionWriter::compression_thread_fn()
 {
+  if (compression_options_.thread_nice_value) {
+
+#ifdef _WIN32
+    ROSBAG2_COMPRESSION_LOG_WARN_STREAM("Lowering of process priority is not implemented for windows");
+
+    /**
+     * The implementation should look something like this
+
+    uint8_t nice_value = *compression_options_.thread_nice_value;
+
+    // this must match THREAD_PRIORITY_IDLE, THREAD_PRIORITY_LOWEST...
+    DWORD dwThreadPri = *compression_options_.thread_nice_value;
+
+    if(!SetThreadPriority(GetCurrentThread(), dwThreadPri))
+    {
+      ROSBAG2_COMPRESSION_LOG_WARN_STREAM(
+        "Could not set nice value of compression thread to " << nice_value << " : " << std::strerror(
+          errno));
+    }
+    */
+
+#else
+    int wanted_nice_value = *compression_options_.thread_nice_value;
+
+    int cur_nice_value = getpriority(PRIO_PROCESS, 0);
+
+    int new_nice_value = nice(wanted_nice_value - cur_nice_value);
+    if(new_nice_value == -1 || new_nice_value != wanted_nice_value)
```

Comment:
I went for
`if ((new_nice_value == -1 && errno != 0) || new_nice_value != wanted_nice_value)`
As this would additionally catch other errors, were new_nice_value != wanted_nice_value, even though errno would not be set.

Reply:
@jmachowinski According to the https://linux.die.net/man/2/nice
> Notes
SUSv2 and POSIX.1-2001 specify that nice() should return the new nice value. However, the Linux syscall and the nice() library function provided in older versions of (g)libc (earlier than glibc 2.2.4) return 0 on success. The new nice value can be found using [getpriority](https://linux.die.net/man/2/getpriority)(2).

Adding `|| new_nice_value != wanted_nice_value` to the condition will not get the advantage of catching any other errors.  

It will affect the portability of such code. One could fork `rosbag2` repo and try to run in with QNX or some other RTOS  and may get in trouble due to this extra check.

I would strongly recommend to not use this check for errors handling.

Reply:
removed